### PR TITLE
Split time controls re-designed

### PR DESCRIPTION
**Pull request by ilaripih**
Re-designed the UI and re-factored some of the code for pull 599, i.e. split time controls. This also includes and thus replaces, everything in pull 599.

### CODE REVIEW

File: projects/gui/src/newtournamentdialog.cpp [hunk 1]
```diff
@@ -207,10 +219,14 @@ void NewTournamentDialog::moveEngine(int offset)
 	int row1 = index.row();
 	int row2 = row1 + offset;
 	EngineConfiguration tmp(m_addedEnginesManager->engineAt(row1));
+	TimeControl tc = m_timeControls.at(row1);
 
 	m_addedEnginesManager->updateEngineAt(row1, m_addedEnginesManager->engineAt(row2));
 	m_addedEnginesManager->updateEngineAt(row2, tmp);
 
+	m_timeControls[row1] = m_timeControls.at(row2);
+	m_timeControls[row2]= tc;
```
Comment (artoj) [inline projects/gui/src/newtournamentdialog.cpp:228]:
```
cutechess/projects/gui/src/newtournamentdialog.cpp: In member function ‘void NewTournamentDialog::moveEngine(int)’:    
cutechess/projects/gui/src/newtournamentdialog.cpp:228:29: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]    
  228 |         m_timeControls[row2]= tc;    
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qbytearray.h:46,    
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qstring.h:50,    
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:47,    
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qwidget.h:45,    
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qdialog.h:44,    
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QDialog:1,    
                 from cutechess/projects/gui/src/newtournamentdialog.h:22,    
                 from cutechess/projects/gui/src/newtournamentdialog.cpp:18:    
/usr/include/x86_64-linux-gnu/qt5/QtCore/qarraydata.h:129:8: note: at offset [50, 74] into destination object ‘QTypedArrayData<TimeControl>::<anonymous>’ of size 24    
  129 | struct QTypedArrayData    
      |        ^~~~~~~~~~~~~~~ 
````

**Comment by ilaripih** [comment]
I'll fix the array offset thing, if I can confirm that there's a bug in Cute Chess code. I'll also fix the dialog types for `TimeControlDialog` and `PairTimeControlDialog` that seem to have changed.

File: projects/gui/src/newtournamentdialog.cpp (hunk 1 above)
Comment (artoj) [reply projects/gui/src/newtournamentdialog.cpp]: Seems this is a false positive; I don't see this error on the CI build logs.

**Comment by artoj** [comment]
LGTM

**Review by artoj — APPROVED** [review]

**Pull request merged**